lib/bigtable/table.js

```diff
@@ -0,0 +1,835 @@
+      return;
+    }
+
+    var family = self.family(resp.name);
+    callback(null, family, resp);
+  });
+};
+
+/**
+ * Delete all rows in the table corresponding to a particular prefix.
+ *
+ * @param {options=} options - Configuration object.
+ * @param {string} options.prefix - Row key prefix, when omitted all rows
+ *     will be deleted.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * //-
+ * // You can supply a prefix to delete all corresponding rows.
+ * //-
+ * var callback = function(err, apiResponse) {
+ *   if (!err) {
+ *     // Rows successfully deleted.
+ *   }
+ * };
+ *
+ * table.deleteRows({
+ *   prefix: 'lincoln.'
+ * }, callback);
+ *
+ * //-
+ * // If you choose to omit the prefix, all rows in the table will be deleted.
+ * //-
+ * table.deleteRows(callback);
+ */
+Table.prototype.deleteRows = function(options, callback) {
+  if (is.function(options)) {
+    callback = options;
+    options = {};
+  }
+
+  var grpcOpts = {
+    service: 'BigtableTableService',
+    method: 'bulkDeleteRows'
+  };
+
+  var reqOpts = {
+    tableName: this.name
+  };
+
+  if (options.prefix) {
+    reqOpts.rowKeyPrefix = Entry.convertToBytes(options.prefix);
+  } else {
+    reqOpts.deleteAllDataFromTable = true;
+  }
+
+  this.request(grpcOpts, reqOpts, function(err, resp) {
+    if (err) {
+      callback(err, resp);
+      return;
+    }
+
+    callback(null, resp);
+  });
+};
+
+/**
+ * Get a reference to a Table Family.
+ *
+ * @param {string} name - The family name.
+ * @return {module:bigtable/family}
+ *
+ * @example
+ * var family = table.family('my-family');
+ */
+Table.prototype.family = function(name) {
+  return new Family(this, name);
+};
+
+/**
+ * Get Family objects for all the column familes in your table.
+ *
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {module:bigtable/family[]} callback.families - The list of families.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * table.getFamilies(function(err, families, apiResponse) {
+ *   // `families` is an array of Family objects.
+ * });
+ */
+Table.prototype.getFamilies = function(callback) {
+  var self = this;
+
+  this.getMetadata(function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    var families = Object.keys(resp.columnFamilies).map(function(familyId) {
+      var family = self.family(familyId);
+      family.metadata = resp.columnFamilies[familyId];
+      return family;
+    });
+
+    callback(null, families, resp);
+  });
+};
+
+/**
+ * Get Row objects for the rows currently in your table.
+ *
+ * @param {options=} options - Configuration object.
+ * @param {string=} options.key - An individual row key.
```

Comment:
I think if `options` is optional, it is implied that everything under that object is optional as well, thus not requiring being marked with a `=`. This is definitely in other parts of our API, but I've been trying to get to them.